src/services/lock/service/lock.service.ts

```diff
@@ -39,29 +37,22 @@ export class LockService {
       .map<[number, string]>((e: LockNode): [number, string] => {
         let distance: number = 0;
         let i = 0;
-        // Ensure both e and start_id are valid, else return a tuple with a high distance value to sort later
-        if (!e || !start_id) return [Number.MAX_SAFE_INTEGER, start_id];
+        if (!e || !start_id || !e.node_id) return [Number.MAX_SAFE_INTEGER, start_id];
         const min = Math.min(start_id.length, e.node_id.length);
         const max = Math.max(start_id.length, e.node_id.length);
         for (; i < min; ++i) {
-          distance = distance * 256 + ((start_id as any)[i] ^ e[i]);
+          distance = distance * 256 + (start_id.charCodeAt(i) ^ e.node_id.charCodeAt(i));
         }
         for (; i < max; ++i) distance = distance * 256 + 255;
         return [distance, e.node_id];
       })
       .sort((a, b) => a[0] - b[0]);
     const date = new Date();
     if (!nodesClosest.length) return start_id;
-    // Safely access the selected node, ensuring we don't accidentally access undefined
     const selectedNode = nodesClosest[date.getMinutes() % nodesClosest.length];
     return selectedNode ? selectedNode[1] : start_id;
   }
 
-  /**
-   * Executes a given task with lock control to prevent concurrent execution.
-   * @param {string} taskId - A unique identifier for the task.
-   * @param {Function} task - A callback function representing the task to be executed.
-   */
   async executeWithLock(taskId: string, task: () => Promise<void>): Promise<void> {
```

Comment:
Function comments are fine, but extra types are not necessary
```suggestion
  /**
   * Executes a given task with lock control to prevent concurrent execution.
   * @param taskId - A unique identifier for the task.
   * @param task - A callback function representing the task to be executed.
   */
  async executeWithLock(taskId: string, task: () => Promise<void>): Promise<void> {
```